File: Semester3/Advanced Methods of Programming/Toy Language Interpreter C#/Assignement7/Assignement7/Model/Expression.cs
using Assignement7.Collection;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Assignement7.Model
{
    public abstract class Expression
    {
        public abstract int Evaluate(MyIDictionary<string, int> symbolTable);
        public abstract override string ToString();
    }
}

File: Semester3/Advanced Methods of Programming/Toy Language Interpreter JavaFx/Assignment8/src/Model/Expression/Expression.java
package Model.Expression;

import Collection.Dictionary.MyIDictionary;
import Model.Exceptions.ADTEmptyException;
import Model.Exceptions.ComparisonExpression;
import Model.Exceptions.DivisionByZero;
import Model.Exceptions.VariableNotFoundException;


public abstract class Expression {
    abstract public Integer evaluate(MyIDictionary<String, Integer> symbolTable, MyIDictionary<Integer, Integer> heapTable) throws ADTEmptyException, DivisionByZero, VariableNotFoundException, ComparisonExpression;
    abstract public String toString();
}
